tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -78,11 +79,13 @@ class EnqueueReadBufferCommand : public Command {
     SystemMemoryWriter& writer;
     void* dst;
     static constexpr EnqueueCommandType type_ = EnqueueCommandType::ENQUEUE_READ_BUFFER;
+    std::optional<uint32_t> core_id;
 
    public:
     Buffer& buffer;
     uint32_t read_buffer_addr;
     EnqueueReadBufferCommand(Device* device, Buffer& buffer, void* dst, SystemMemoryWriter& writer);
+    EnqueueReadBufferCommand(Device* device, Buffer& buffer, void* dst, SystemMemoryWriter& writer, const uint32_t & core_id);
```

Comment:
I noticed that we never use this variant of the constructor. Can we remove it? In enqueue_read_buffer, we only ever do this

```c++
EnqueueReadBufferCommand command(this->device, buffer, dst, this->sysmem_writer);
```